Detectors/MUON/MCH/Simulation/src/Digitizer.cxx

```diff
@@ -157,12 +160,53 @@ int Digitizer::processHit(const Hit& hit, int detID, double event_time)
       q *= chargenon;
     }
     auto signal = resp.response(q);
-    digits.emplace_back(padid, signal);
+    digits.emplace_back(time, padid, signal);
+    multimap.emplace(padid, digits.size());
```

Comment:
I would leave that forEachPadInArea loop untouched, and handle the digit merging logic only in the mergeDigits function, to keep things simple.